### PR TITLE
[ML-9511] create model registry link in the main UI

### PR DESCRIPTION
**Pull request by zhidongqu-db**
## What changes are proposed in this pull request?

create model registry link in the main UI

## How is this patch tested?

manual test from different pages navigating to the header model registry UI link

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The user will be able to navigate to model registry page through the link the the header bar

Screen Shot 2020-05-08 at 3 40 36 PM

Screen Shot 2020-05-08 at 3 40 46 PM

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by Zangr — COMMENTED** [review]
@zhidongqu-db Nice work! 

Let's fix the GitHub action related recommendation first. To automatically remove trailing space, you may want to set your IntelliJ accordingly:
`Preference → Editor → General → Strip trailing spaces on Save` => All

**Comment by zhidongqu-db** [comment]
> @zhidongqu-db Nice work!
> 
> Let's fix the GitHub action related recommendation first. To automatically remove trailing space, you may want to set your IntelliJ accordingly:
> `Preference → Editor → General → Strip trailing spaces on Save` => All

Fixed styling issues!

File: mlflow/server/js/src/experiment-tracking/components/App.js [hunk 1]
```diff
@@ -36,6 +36,31 @@ class App extends Component {
                   <img className='mlflow-logo' alt='MLflow' src={logo} />
                 </Link>
               </div>
+              <div className='header-route-links'>
+                <NavLink
+                  strict
+                  to={Routes.rootRoute}
+                  activeStyle={{ borderBottom: '4px solid #43C9ED' }}
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.js:43]: Let's externalize this to a style constant below the file and use it here:
```
const classNames = {
  activeNavLink: { borderBottom: '4px solid #43C9ED' },
}
```

File: mlflow/server/js/src/experiment-tracking/components/App.js [hunk 2]
```diff
@@ -36,6 +36,31 @@ class App extends Component {
...
+                  className='header-nav-link-experiments'
+                >
+                  <div className='experiments'>
+                    <span>Experiments</span>
+                  </div>
+                </NavLink>
+                <NavLink
+                  strict
+                  to={modelListPageRoute}
+                  activeStyle={{ borderBottom: '4px solid #43C9ED' }}
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.js:56]: ditto

File: mlflow/server/js/src/experiment-tracking/components/App.js [hunk 3]
```diff
@@ -36,6 +36,31 @@ class App extends Component {
...
                 </Link>
               </div>
+              <div className='header-route-links'>
+                <NavLink
+                  strict
+                  to={Routes.rootRoute}
+                  activeStyle={{ borderBottom: '4px solid #43C9ED' }}
+                  isActive={(match, location) => {
+                    return match && !location.pathname.includes('models');
+                  }}
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.js:46]:
```suggestion
                  isActive={(match, location) => match && !location.pathname.includes('models')}
```

File: mlflow/server/js/src/experiment-tracking/components/App.css [hunk 4]
```diff
@@ -92,6 +92,22 @@ div.docs {
   color: #ffffff;
 }
 
+.header-nav-link-experiments {
+  display: inline-block;
+  height: 39px;
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.css:97]: According to @gioa 's new design, we are currently keeping all px based measures a multiple of 4 when possible.

File: mlflow/server/js/src/experiment-tracking/components/App.css [hunk 5]
```diff
@@ -100,6 +116,13 @@ div.header-links {
   font-size: 16px;
 }
 
+div.header-route-links {
+  display: inline-block;
+  padding-top: 21px;
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.css:121]: multiple of 4

File: mlflow/server/js/src/experiment-tracking/components/App.css [hunk 6]
```diff
@@ -92,6 +92,22 @@ div.docs {
...
 
+.header-nav-link-experiments {
+  display: inline-block;
+  height: 39px;
+}
+
+.header-nav-link-models {
+  display: inline-block;
+  margin-left: 24px;
+  height: 39px;
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.css:103]: multiple of 4

File: mlflow/server/js/src/experiment-tracking/components/App.css [hunk 7]
```diff
@@ -92,6 +92,22 @@ div.docs {
...
+}
+
+.experiments, .models {
+  display: inline-block;
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/App.css:107]: `inline-block` mandates a font-size related gap between divs. Our current convention is to use `display: flex` on the parent div to get the block-level element horizontal laid out.

**Review by Zangr — COMMENTED** [review]
Look great overall! I recommended some small changes and nits.

**Review by Zangr — APPROVED** [review]
LGTM!

**Review by dbczumar — APPROVED** [review]
LGTM! Awesome work @zhidongqu-db!

**Pull request merged**